packages/neon_framework/lib/src/storage/shared_preferences_persistence.dart

```diff
@@ -0,0 +1,110 @@
+// ignore_for_file: cascade_invocations
+
+import 'package:meta/meta.dart';
+import 'package:neon_framework/src/storage/persistence.dart';
+import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
+import 'package:shared_preferences_platform_interface/types.dart';
+
+/// An SharedPreferences backed cached persistence for preferences.
+///
+/// There is only one cache backing all `SharedPreferencesPersistence`
+/// instances. Use the [prefix] to separate different storages.
+/// Keys within a storage must be unique.
+@internal
+final class SharedPreferencesPersistence extends CachedPersistence {
+  /// Creates a new SharedPreferences persistence.
+  SharedPreferencesPersistence({String prefix = ''}) : _prefix = prefix;
+
+  static SharedPreferencesStorePlatform get _store => SharedPreferencesStorePlatform.instance;
+
+  /// The prefix of this persistence.
+  ///
+  /// Keys within it must be unique.
+  @protected
+  String get prefix {
+    if (_prefix.isEmpty) {
+      return 'flutter.';
+    }
+
+    return 'flutter.$_prefix-';
+  }
+
+  final String _prefix;
+
+  @override
+  Map<String, Object> get cache {
+    if (_globalCache.containsKey(prefix)) {
+      return _globalCache[prefix]! as Map<String, Object>;
+    }
+
+    final cache = <String, Object>{};
+    final keys = _globalCache.keys;
+    for (final prefixedKey in keys) {
+      if (prefixedKey.startsWith(prefix)) {
+        final key = prefixedKey.replaceFirst(prefix, '');
+        cache[key] = _globalCache[prefixedKey]!;
+      }
+    }
+
+    _globalCache[prefix] = cache;
+    return cache;
+  }
+
+  static final Map<String, Object> _globalCache = {};
+
+  /// Initializes all persistences by setting up the backing SharedPreferences
+  /// storage and priming the global cache.
+  ///
+  /// This must be called and completed before any sqlite persistence.
```

Comment:
Why is sqlite mentioned here?

Reply:
I initially wrote the persistence with https://github.com/nextcloud/neon/pull/1585 in mind and just later backported it to the current shared_prefs. 
So basically copy pasta.